test/parallel/test-child-process-spawnsync-validation-errors.js

```diff
@@ -0,0 +1,176 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+const spawnSync = require('child_process').spawnSync;
+
+function pass(option, value) {
+  // Run the command with the specified option. Since it's not a real command,
+  // spawnSync() should run successfully but return an ENOENT error.
+  const child = spawnSync('not_a_real_command', { [option]: value });
+
+  assert.strictEqual(child.error.code, 'ENOENT');
+}
+
+function fail(option, value, message) {
+  assert.throws(() => {
+    spawnSync('not_a_real_command', { [option]: value });
+  }, message);
+}
+
+{
+  // Validate the cwd option
+  const err = /TypeError: "cwd" must be a string/;
+
+  pass('cwd', undefined);
+  pass('cwd', null);
+  pass('cwd', __dirname);
+  fail('cwd', 0, err);
+  fail('cwd', 1, err);
+  fail('cwd', true, err);
+  fail('cwd', false, err);
+  fail('cwd', [], err);
+  fail('cwd', {}, err);
```

Comment:
Here and below I would add Function to the `fail()` cases (demonstrating it doesn't get coerced to a string).